core/src/main/kotlin/org/evomaster/core/output/naming/ActionTestCaseNamingStrategy.kt

```diff
@@ -51,6 +62,34 @@ abstract class ActionTestCaseNamingStrategy(
         } else {
             addActionResult(individual.evaluatedMainActions().last(), nameTokens)
         }
+        addEnvironmentActions(individual, nameTokens)
+    }
+
+    private fun addEnvironmentActions(individual: EvaluatedIndividual<*>, nameTokens: MutableList<String>) {
+        val initializingActions = individual.individual.seeInitializingActions()
+        val allActions = individual.individual.seeAllActions()
+
+        val initActionNames = mutableListOf<String>()
+        if (hasSqlAction(initializingActions)) initActionNames.add(sql)
+        if (hasMongoAction(initializingActions)) initActionNames.add(mongo)
+        if (usesWireMock(allActions)) initActionNames.add(wiremock)
+
+        if (initActionNames.isNotEmpty()) {
+            nameTokens.add(using)
+            nameTokens.addAll(initActionNames)
+        }
+    }
+
+    private fun hasSqlAction(actions: List<EnvironmentAction>): Boolean {
+        return actions.any { it is SqlAction }
+    }
+
+    private fun hasMongoAction(actions: List<EnvironmentAction>): Boolean {
+        return actions.any { it is MongoDbAction }
+    }
+
+    private fun usesWireMock(actions: List<Action>): Boolean {
+        return config.isEnabledExternalServiceMocking() && actions.any { it is HostnameResolutionAction }
```

Comment:
there are 2 different types of actions for handling external services `HostnameResolutionAction` and `HttpExternalServiceAction`. the latter is related to WireMock.

Reply:
My bad, fixed!